SignalMessaging/attachments/SignalAttachment.swift

```diff
@@ -746,20 +753,61 @@ public class SignalAttachment: NSObject {
         }
     }
 
-    private class func imageScaled(_ image: UIImage, toMaxSize size: CGFloat) -> UIImage {
+    // NOTE: For unknown reasons, resizing images with UIGraphicsBeginImageContext()
+    // crashes reliably in the share extension after screen lock's auth UI has been presented.
+    // Resizing using a CGContext seems to work fine.
+    private class func imageScaled(_ uiImage: UIImage, toMaxSize size: CGFloat) -> UIImage? {
+        guard let cgImage = uiImage.cgImage else {
+            owsFail("\(logTag) UIImage missing cgImage.")
+            return nil
+        }
+
+        // It's essential that we work consistently in "CG" coordinates (which are
+        // and pixels and don't reflect orientation), not "UI" coordinates (which
+        // are in points and do reflect orientation).
+        let srcWidth = CGFloat(cgImage.width)
+        let srcHeight = CGFloat(cgImage.height)
         var scaleFactor: CGFloat
-        let aspectRatio: CGFloat = image.size.height / image.size.width
+        let aspectRatio: CGFloat = srcWidth / srcHeight
```

Comment:
So previously AR was height/width but now it's width/height.

Is that because of the CG Coord-> UI Coord  switch?

Reply:
Nope, that was a typo.  I'm going to switch to using `AVMakeRectWithAspectRatioInsideRect()` to do this math to avoid this kind of thing.

Reply:
Nice, didn't know about that one.